### PR TITLE
feat: correct realize building unit backoffice

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/BuildingUnit/CorrectBuildingUnitRealizationLambdaRequest.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public sealed class CorrectBuildingUnitRealizationLambdaRequest :
+        BuildingUnitLambdaRequest,
+        IHasBackOfficeRequest<BackOfficeCorrectBuildingUnitRealizationRequest>,
+        IHasBuildingUnitPersistentLocalId
+    {
+        public BackOfficeCorrectBuildingUnitRealizationRequest Request { get; set; }
+
+        public int BuildingUnitPersistentLocalId => Request.BuildingUnitPersistentLocalId;
+
```
Comment (ArneD) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/BuildingUnit/CorrectBuildingUnitRealizationLambdaRequest.cs:16]: please add a `ctor` which accepts `BackOfficeCorrectBuildingUnitRealizationRequest`

File: src/BuildingRegistry.Api.BackOffice.Abstractions/Building/Validators/ValidationErrors.cs [hunk 2]
```diff
@@ -25,9 +25,11 @@ public static class BuildingUnit
             public const string BuildingNotFound = "GebouweenheidGebouwIdNietGekendValidatie";
             public const string BuildingStatusNotInRealized = "GebouwStatusNietInGerealiseerd";
             public const string BuildingUnitCannotBeRealized = "GebouweenheidGehistoreerdOfNietGerealiseerd";
+            public const string BuildingUnitCannotBeCorrectedFromRealized = "GebouweenheidGeplandOfNietgerealiseerdOfGehistoreerd";
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Abstractions/Building/Validators/ValidationErrors.cs:28]: I think this should become "GebouweenheidNietgerealiseerdOfGehistoreerd". Planned is the desired state and we don't include it in the error code.

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/CorrectBuildingUnitRealizationLambdaHandler.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        protected override TicketError? MapDomainException(DomainException exception, CorrectBuildingUnitRealizationLambdaRequest request)
+        {
+            return exception switch
+            {
+                BuildingUnitHasInvalidFunctionException => new TicketError(
+                    ValidationErrorMessages.BuildingUnit.BuildingUnitHasInvalidFunction,
+                    ValidationErrorCodes.BuildingUnit.BuildingUnitHasInvalidFunction),
+                BuildingUnitHasInvalidStatusException => new TicketError(
+                    ValidationErrorMessages.BuildingUnit.BuildingUnitStatusNotInRealized,
+                    ValidationErrorCodes.BuildingUnit.BuildingUnitCannotBeCorrectedFromRealized),
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/CorrectBuildingUnitRealizationLambdaHandler.cs:66]: Can we keep the variable name of the error code and error message the same?

File: src/BuildingRegistry.Projections.Legacy/BuildingSyndication/BuildingSyndicationProjections.cs [hunk 4]
```diff
@@ -971,6 +971,16 @@ await context
                 }, ct);
             });
 
+             When<Envelope<BuildingUnitWasCorrectedFromRealizedToPlanned>>(async (context, message, ct) =>
+            {
+                await context.CreateNewBuildingSyndicationItem(message.Message.BuildingPersistentLocalId, message, item =>
+                {
+                    var unit = item.BuildingUnitsV2.Single(y => y.PersistentLocalId == message.Message.BuildingUnitPersistentLocalId);
+                    unit.Status = BuildingRegistry.Building.BuildingUnitStatus.Realized;
```
Comment (jvandaal) [inline src/BuildingRegistry.Projections.Legacy/BuildingSyndication/BuildingSyndicationProjections.cs:979]: Status should be Planned

File: test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitRealization/GivenBuildingUnitPlanned.cs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    using Building.Exceptions;
+    using BuildingRegistry.Api.BackOffice.Abstractions.BuildingUnit.Validators;
+    using FluentAssertions;
+    using FluentValidation;
+    using Microsoft.AspNetCore.Http;
+    using Moq;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class GivenBuildingUnitPlanned : BackOfficeApiTest
```
Comment (jvandaal) [inline test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitRealization/GivenBuildingUnitPlanned.cs:26]: Rename class to GivenBuildingUnitRealized

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/BuildingUnit/CorrectBuildingUnitRealizationLambdaRequest.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        /// <summary>
+        /// Map to CorrectBuildingUnitRealization command
+        /// </summary>
+        /// <returns>RealizeBuildingUnit.</returns>
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/BuildingUnit/CorrectBuildingUnitRealizationLambdaRequest.cs:20]: Change returns comment

**Review by jvandaal — CHANGES_REQUESTED** [review]

**Pull request merged**